File: pe-math/src/main/java/org/saar/maths/noise/MultipliedNoise3f.java
package org.saar.maths.noise;

public class MultipliedNoise3f implements Noise3f {

    private final int multiply;
    private final Noise3f noise3f;

    public MultipliedNoise3f(int multiply, Noise3f noise3f) {
        this.multiply = multiply;
        this.noise3f = noise3f;
    }

    @Override
    public float noise(float x, float y, float z) {
        return this.noise3f.noise(x, y, z) * this.multiply;
    }
}

File: pe-math/src/main/kotlin/org/saar/maths/noise/MultipliedNoise3f.kt
package org.saar.maths.noise

class MultipliedNoise3f(private val multiply: Float, private val noise3f: Noise3f) : Noise3f {

    override fun noise(x: Float, y: Float, z: Float) = this.noise3f.noise(x, y, z) * this.multiply
}
